### PR TITLE
bump bufr version and remove wcoss_dell_p3 references

### PR DESCRIPTION
**Pull request by aerorahul**
This PR:
- updates bufr library to version 11.7.0
- flushes CI to rebuild cache
- removes references to wcoss_dell_p3 as that machine is decommissioned.

### CODE REVIEW

File: .github/workflows/gcc.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ defaults:
     shell: bash -leo pipefail {0}
 
 env:
-  cache_key: gcc3  # The number (#) following the cache_key "gcc" is to flush Action cache.
+  cache_key: gcc4  # The number (#) following the cache_key "gcc" is to flush Action cache.
```
Comment (RussTreadon-NOAA) [inline .github/workflows/gcc.yml:11]: Just for my learning purposes, why do we increase gcc from 3 to 4?   Similar question for intel in intel.yml.

File: INSTALL.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ NCEP Libraries (NCEPLibs) compiled with the same compiler and MPI library (where
 
 | Library         | Supported (tested) versions                                |
 |-----------------|------------------------------------------------------------|
-| BUFR            | 11.4.0 and above                                           |
+| BUFR            | 11.6.0 and above                                           |
```
Comment (RussTreadon-NOAA) [inline INSTALL.md]: Why not make this 11.7.0 and above?

**Review by RussTreadon-NOAA — COMMENTED** [review]
Looks good.  One _"I'm trying to learn"_ question and one _"why not?"_ question.    I can approve after hearing back on the _"why not?"_ question.

File: .github/workflows/gcc.yml (hunk 1 above)
Comment (aerorahul) [reply .github/workflows/gcc.yml:11]: Good question.
There is a comment in this yaml file (reproduced below):
```
# A note on flushing Action cache and relevance to "cache_key" above.
# There is no way to flush the Action cache, and hence a number (#) is appended
# to the "cache_key" (gcc).
# If the dependencies change, increment this number and a new cache will be
# generated by the dependency build step "setup"
# There is a Github issue to force clear the cache.
# See discussion on:
# https://stackoverflow.com/questions/63521430/clear-cache-in-github-actions
```
Basically, if we change the version of the dependencies, we need to rebuild the cache for it to take effect.
The cache is then retained and reused in subsequent PR's if there is no update in the dependencies, saving time in the CI.  
We don't update the library versions often, so there is no need to touch this file (and increment the key) with every PR.
But when we do update, this cache needs to be rebuilt once.

File: INSTALL.md (hunk 2 above)
Comment (aerorahul) [inline INSTALL.md]:
```suggestion
| BUFR            | 11.7.0 and above                                           |
```

Comment (aerorahul) [reply INSTALL.md]: done.

Comment (RussTreadon-NOAA) [reply INSTALL.md]: Thanks, @aerorahul

File: .github/workflows/gcc.yml (hunk 1 above)
Comment (RussTreadon-NOAA) [reply .github/workflows/gcc.yml:11]: OK.  Let me digest this.   Thank you, @aerorahul

**Review by RussTreadon-NOAA — APPROVED** [review]
Approve.

**Pull request merged**